pfcpiface/main.go

```diff
@@ -77,7 +409,16 @@ func main() {
 	}
 	defer c.ResumeAll(ctx, &pb.EmptyRequest{})
 
-	// create and add pdr
-	p := pdr{}
-	addPDR(ctx, c, p)
+	for i := uint32(0); i < conf.MaxSessions; i++ {
+		// create and add pdr
+		addPDR(ctx, c, uint32(i))
+
+		// create and add far
+		addFAR(ctx, c, s1u_sgw_ip, uint32(i))
+
+		// create and add counters
+		addCounters(ctx, c, uint32(i))
```

Comment:
make these functions' prototypes what they need to be to accept everything they need to perform the function. Make them just like cpiface/bess_control.h. Define structures that they care about and pass them. Any hardcoding to test sim should be outside. You can even split this file just like cpiface. main.go and bess.go.